### PR TITLE
framework/task_manager: Modify naming and internal fucntion.

### PR DESCRIPTION
**Pull request by tyha1115**
framework/task_manager: Modify file name and declared variable names.

In order to reduce the opportunity of misreadings, file name and several variable names were changed.
1. File name change
 - include/tinyara/task_manager_internal.h -> include/tinyara/task_manager_drv.h
2. Variable name chage
 - typedef of callback (_tm_unicast_t -> tm_unicast_callback_t / _tm_broadcast_t -> tm_broadcast_callback_t / _tm_termination_t -> tm_termination_callback_t)

framework/task_manager: Modify internal function (taskmgr_getinfo_with_name).

According to the type of task/thread (built_in task, task, pthread), the lists that stores name of task/thread are different.
Therefore, an invalid referencing can occur which can result in an assert problem.
The procedure of selecting the lists to find the name of task/thread was added according to the type.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [87bb3405b844e5a6c3275fb48fd62b81bff30b02] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [87bb3405b844e5a6c3275fb48fd62b81bff30b02] - Code Rule Check (C++) OK.

File: framework/src/task_manager/task_manager_core.c [hunk 1]
```diff
@@ -754,18 +754,35 @@ static int taskmgr_getinfo_with_name(char *name, tm_response_t *response_msg)
 	response_msg->data = NULL;
 
 	for (chk_idx = 0; chk_idx < CONFIG_TASK_MANAGER_MAX_TASKS; chk_idx++) {
-		if (TM_LIST_ADDR(chk_idx) && ((strncmp(builtin_list[TM_IDX(chk_idx)].name, name, strlen(name)) == 0)
-					   || (strncmp(tm_task_list[TM_IDX(chk_idx)].name, name, strlen(name)) == 0)
+		if (TM_LIST_ADDR(chk_idx)) {
```
Comment (sunghan-chang) [inline framework/src/task_manager/task_manager_core.c]: Please refactor this logic like below:
```
char *thread_name = NULL;

if (builtin) {
   thread_name = builtin_list[ ].name;
} else if (task) {
   thread_name = tm_task_list[ ].name;
} else if (pthread) {
   thread_name = tm_pthread_list[ ].name;
}

if (thread_name && !strncmp(thread_name, name, strlen(name))) {
   ....
}
```

Comment (sunghan-chang) [reply framework/src/task_manager/task_manager_core.c]: Don't you need to use `strlen(name) + 1`?

**Comment by seinfra** [comment]
Target : [e58caad28d2949c1542cda8d57c80fd9418b62a2] -

**Comment by seinfra** [comment]
Target : [1d82f64c24ad57f11c823390d393cbe62d13817c] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [1d82f64c24ad57f11c823390d393cbe62d13817c] - Code Rule Check (C++) OK.

File: framework/include/task_manager/task_manager.h [hunk 2]
```diff
@@ -133,17 +133,17 @@ typedef struct tm_msg_s tm_msg_t;
 /**
  * @brief Unicast callback function type
  */
-typedef void (*_tm_unicast_t)(tm_msg_t *);
+typedef void (*tm_unicast_callback_t)(tm_msg_t *);
```
Comment (sunghan-chang) [inline framework/include/task_manager/task_manager.h]: The set_unicast_cb API does not use function pointer. Why do you expose this type?
There are two options.
option 1. Let's move this into task_manager_internal.h. 
option 2. Let's use this type in set_unicast_cb API like below:
    `int task_manager_set_unicast_cb(tm_unicast_callback_t cb_func);`

File: framework/src/task_manager/task_manager_core.c (hunk 1 above)
Comment (tyha1115) [reply framework/src/task_manager/task_manager_core.c]: Done.

**Comment by seinfra** [comment]
Target : [09bdd06c04701466ca9320f288069a25cfedaa6b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [09bdd06c04701466ca9320f288069a25cfedaa6b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6a822bc2b72a957faaa49e6d79116f3a199167b1] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6a822bc2b72a957faaa49e6d79116f3a199167b1] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [02c7bca383fab56efffec5255371c4001cc0f4cc] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [02c7bca383fab56efffec5255371c4001cc0f4cc] - Code Rule Check (C++) OK.

**Pull request merged**